core/drivers/stm32_i2c.c

```diff
@@ -824,16 +814,14 @@ int stm32_i2c_init(struct i2c_handle_s *hi2c,
 	hi2c->i2c_err = I2C_ERROR_NONE;
 	hi2c->i2c_state = I2C_STATE_READY;
 
-	rc = i2c_config_analog_filter(hi2c, init_data->analog_filter);
-	if (rc)
-		DMSG("I2C analog filter error %d", rc);
+	i2c_config_analog_filter(hi2c, init_data->analog_filter);
 
 	if (IS_ENABLED(CFG_STM32MP13))
 		stm32_pinctrl_set_secure_cfg(hi2c->pinctrl, true);
 
 	clk_disable(hi2c->clock);
 
-	return rc;
+	return 0;
```

Comment:
This makes me realize that stm32_i2c_init() returns an int and that the probe function uses the return code as a TEE_Result, can you please make the change?

The exported function are also in the same state but don't require a change for now

Reply:
I'll append a fixup commit for `stm32_i2c_probe()`.